ship.java

```diff
@@ -0,0 +1,18 @@
+public class ship {
+    private List<KeyVersion> processEventsInBucket(String bucketName) {
+      final AmazonS3 s3Client = EventHandler.getS3Client();
+      ObjectListing files = s3Client.listObjects(bucketName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code uses an outdated API. [ListObjectsV2 is the revised List Objects API, and we recommend you use this revised API for new application developments](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html).